src/plugins/resolver/resolver.c

```diff
@@ -68,6 +80,16 @@ static void resolverInit (resolverHandle *p, const char *path)
 	p->tempfile = 0;
 
 	p->path = path;
+
+	{
+		pthread_mutexattr_t mutex_attr;
+
+		int result = pthread_mutexattr_init(&mutex_attr);
+		if(result == 0)
+			result = pthread_mutexattr_settype(&mutex_attr, PTHREAD_MUTEX_RECURSIVE);
+		if(result == 0)
+			result = pthread_mutex_init(getResolverMutex(), &mutex_attr);
```

Comment:
Its not that simple, int is not thread-safe. So we would need a mutex to protect the recursive mutex initialization. If PTHREAD_MUTEX_RECURSIVE works, its the ideal solution. Otherwise (if no such thing for Apple exists or works) I would like to have the double-mutex thing *only* for Apple or other OS not supporting proper recursive mutex initialization.